src/Presentation/Views/Registration/RegistrationViewModel.ts

```diff
@@ -1,33 +1,109 @@
+    const [registerUsername, setRegisterUsername] = useState("");
+    const [isEmailTaken, setIsEmailTaken] = useState(false);
+    const [isSubmitted, setIsSubmitted] = useState(false);
+
+    const [passwordVisible, setPasswordVisible] = useState(false);
+    const [passwordConfirmationVisible, setPasswordConfirmationVisible] = useState(false);
+    const [passwordType, setPasswordType] = useState(InputType.PASSWORD);
+    const [passwordConfirmationType, setPasswordConfirmationType] = useState(InputType.PASSWORD);
+    
+
+    // Initial values to be set in the form
+    const initialValues = {
+        email: "",
+        emailConfirmation: "",
+        username: "",
+        password: "",
+        passwordConfirmation: ""
+    }
+
+    // Validation criteria for each field in the form
+    const validationSchema = Yup.object().shape({
+        email: Yup.string().email("Please enter a valid email address.").required("Email is required."),
+        emailConfirmation: Yup.string().oneOf([Yup.ref("email"), null], "Email does not match!").required("Please confirm your email address."),
+        username: Yup.string().max(30, "Too Long! Maximum 30 characters.").required("Username is required."),
+        password: Yup.string().min(8, 'Must be at least 8 characters long' ).required("Please provide a password.").matches(/^(?=.*[A-Z])(?=.*[()@#$%^&+=]).*$/, "Password must have at least one capital letter and one special character."),
+        passwordConfirmation: Yup.string().oneOf([Yup.ref('password'), null], "Passwords do not match!").required("Please confirm your password.")
+    });
+
+    // Registering the user
+    async function onClickRegister(formData: NewUser) {
+        const newEmail = formData.email;
+        const newUsername = formData.username;
+        const newPassword = formData.password;
+
+        setRegisterEmail(newEmail);
+        console.log(formData);
+
+        setRegisterUsername(newUsername);
+        console.log(formData);
 
-    async function onClickRegister() {
         try {
             //Creates an account and logins automatically
-            const user = await RegisterUserUseCase(registerEmail, registerPassword);
+            const user = await RegisterUserUseCase(newEmail, newUsername, newPassword);
             console.log(user);
+            setIsSubmitted(true);
         } catch (error: any) {
-            console.log(error.message);
+            console.log(error);
+            console.log(error.code);
+
+            if (error.code === "auth/email-already-in-use") {
+                setIsEmailTaken(true);
+            } else {
+                setIsEmailTaken(false);
+            }
         }
     };
 
-    function onChangeEmail(event: any) {
-        setRegisterEmail(event.target.value);
+    // OnClik methods for making the passwords visible to the user
+    function onClickShowPassword() {
+        if (!passwordVisible) {
+            setPasswordType(InputType.TEXT);
```

Comment:
does this ever need to be text? what does that do relative to when you set it as type password?

Reply:
The type property just refers to how the input is displayed. So type="text" will just be the normal text field and type="password" will hide the text as dots.

![image](https://user-images.githubusercontent.com/65355965/196064596-21f14e9a-ab2c-4fdb-a4b6-c874d98a3420.png)
![image](https://user-images.githubusercontent.com/65355965/196064618-0438df9b-4353-4e81-abd4-1877c163ebee.png)
___

![image](https://user-images.githubusercontent.com/65355965/196064632-8fa36ec9-9e34-4202-8012-b0e97625dfc9.png)
![image](https://user-images.githubusercontent.com/65355965/196064644-0f709ff7-f219-4c0c-8f20-374e6b5521ed.png)

Reply:
What is the purpose for the subsequent line then, can it just always be type password then you only flip the single is visible property?

![image](https://user-images.githubusercontent.com/77797048/196064788-698920de-7024-456d-82a8-b2e7166efc9c.png)


feel free to merge just was wondering on this

Reply:
Ohhh it's probably the checkbox?

Reply:
Yeah, it was a way to keep track of the status of the checkbox, but I think you might be right. I probably could just check what the current password type is and then set it to the opposite.